sda-download/api/sda/sda_test.go

```diff
@@ -552,3 +566,217 @@ func TestEncrypted_Coords(t *testing.T) {
+
+	bufferWriter := bytes.Buffer{}
+	dataWriter, err := streaming.NewCrypt4GHWriter(&bufferWriter, config.Config.App.Crypt4GHPrivateKey, readerPublicKeyList, nil)
+	assert.NoError(t, err, "Could not make crypt4gh writer for test")
+
+	for i := 0; i < 1000; i++ {
+		_, err = dataWriter.Write([]byte("data"))
+		assert.NoError(t, err, "Could not write to crypt4gh writer for test")
+	}
+	dataWriter.Close()
+
+	t.Logf("buffer: %v", bufferWriter.Len())
+	headerBytes, err := headers.ReadHeader(&bufferWriter)
+	assert.NoError(t, err, "Could not get header")
+
+	_, err = io.Copy(datafile, &bufferWriter)
+	assert.NoError(t, err, "Could not write temporary file")
+	datafile.Close()
+
+	// Save original to-be-mocked functions
+	originalCheckFilePermission := database.CheckFilePermission
+	originalGetCacheFromContext := middleware.GetCacheFromContext
+	originalGetFile := database.GetFile
+	archive := config.Config.Archive
+	archive.Posix.Location = "."
+	Backend, _ = storage.NewBackend(archive)
+
+	// Substitute mock functions
+	database.CheckFilePermission = func(_ string) (string, error) {
+		return "dataset1", nil
+	}
+	middleware.GetCacheFromContext = func(_ *gin.Context) session.Cache {
+		return session.Cache{
+			Datasets: []string{"dataset1"},
+		}
+	}
+	database.GetFile = func(_ string) (*database.FileDownload, error) {
+		fileDetails := &database.FileDownload{
+			ArchivePath: datafileName,
+			ArchiveSize: 0,
+			Header:      headerBytes,
+		}
+
+		return fileDetails, nil
+	}
+
+	// Mock request and response holders
+	w := httptest.NewRecorder()
+	c, _ := gin.CreateTestContext(w)
+	c.Request = &http.Request{Method: "GET", URL: &url.URL{}}
+
+	// Test the outcomes of the handler
+	Download(c)
+	response := w.Result()
+	defer response.Body.Close()
+	body, _ := io.ReadAll(response.Body)
+
+	assert.Equal(t, 200, response.StatusCode, "Unexpected status code from download")
+	assert.Equal(t, []byte("data"), body[:4], "Unexpected body from download")
+
+	// Mock request and response holders
+	w = httptest.NewRecorder()
+	c, _ = gin.CreateTestContext(w)
+	c.Request = &http.Request{Method: "GET", URL: &url.URL{Path: "/somepath", RawQuery: "startCoordinate=5&endCoordinate=10"}}
+	// Test the outcomes of the handler
+	Download(c)
+	response = w.Result()
+	defer response.Body.Close()
+	body, _ = io.ReadAll(response.Body)
+
+	assert.Equal(t, 200, response.StatusCode, "Unexpected status code from download")
+	assert.Equal(t, []byte("atada"), body, "Unexpected body from download")
+
+	w = httptest.NewRecorder()
+	c, _ = gin.CreateTestContext(w)
+	c.Request = &http.Request{Method: "GET", URL: &url.URL{Path: "/s3-encrypted/somepath", RawQuery: "filename=somepath"}}
+	c.Request.Header = http.Header{"Client-Public-Key": []string{config.Config.App.Crypt4GHPublicKeyB64},
+		"Range": []string{"bytes=0-10"}}
+
+	c.Params = make(gin.Params, 1)
+	c.Params[0] = gin.Param{Key: "type", Value: "encrypted"}
+
+	// Test the outcomes of the handler
```

Comment:
Maybe this comment can be more informative?
So as the rest of comments `// Test the outcomes of the handler`

Reply:
Yes, I've updated/added comments a bit now, good?